doc/lsst.pipe.base/creating-a-command-line-task.rst

```diff
@@ -0,0 +1,253 @@
+.. NOTE: this topic should be edited into the modernized topic-based documentation style.
+
+.. _creating-a-command-line-task:
+
+############################
+Creating a command-line task
+############################
+
+This document describes how to write a command-line task, which is the LSST version of a complete data processing pipeline.
+To create a command-line task you will benefit from some background:
+
+- Read :doc:`task-framework-overview` to get a basic idea of what tasks are and what classes are used to implement them.
+- Read :doc:`creating-a-task`.
+  A command-line task is an enhanced version of a regular task.
+  Thus all the considerations for writing a normal task also apply to writing a command-line task, and these will not be repeated in this manual.
+- Acquire a basic understanding of data repositories and how to use the butler to read and write data (to be written; for now read existing tasks to see how it is done).
+
+.. _creating-a-command-line-task-intro:
+
+Introduction
+============
+
+A command-line task is an enhanced version of a regular task (see :ref:`creating-a-task`).
+Regular tasks are only intended to be used as relatively self-contained stages in data processing pipelines, whereas command-line tasks can also be used as complete pipelines.
+As such, command-line tasks include :ref:`run scripts <creating-a-command-line-task-run-script>` that run them as pipelines.
+
+Command-line tasks have the following key attributes, in addition to the attributes for regular tasks:
+
+- They are subclasses of `lsst.pipe.base.CmdLineTask`, whereas regular tasks are subclasses of `lsst.pipe.base.Task`.
+- They have an associated :ref:`run script <creating-a-command-line-task-run-script>` to run them from the command-line as pipelines (this is common, but not required, for regular tasks).
+- They have a ``run`` method which performs the full pipeline data processing.
+- By default the ``run`` method takes exactly one argument: a data reference for the item of data to be processed.
```

Comment:
This is wrong (although commonly adhered to, and the rules were not clearly laid down).

A `run` method *may not* take a dataRef; a `runDataref` must be provided to unpack the dataRef and pass the arguments to the `run` method (there are some exceptions for things like reference catalogues where a proxy may be passed).

Because this rule is often/usually ignored I think we have to be a bit careful here.  The docs should describe the correct situation, and tickets filed to fix all offending tasks.